mtg_card_identifier/RootView.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  RootView.swift
+//  mtg_card_identifier
+//
+//  Created by Dylan Martin on 8/11/23.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+struct RootViewReducer: Reducer {
+    struct State: Equatable {
+        var isDatabaseDownloaded = false
+    }
```

Comment:
Instead of using `DownloadDatabaseFilesView_Previews.store` you would have a value of `var downloadDatabaseState = DownloadDatabaseFilesViewReducer.State` (or whatever that state type is) as a variable on State here. 

Then you'd add an Action case:
`case downloadDatabaseFilesViewAction(DownloadDatabaseFilesViewReducer.Action)`

Then down below you would reference it below when you create the view by saying:
`DownloadDatabaseFilesView(store: self.store.scope(RootViewReducer.Action.downloadDatabaseFilesViewAction, \.downloadDatabaseState))` (roughly you'd do something like that. I forget exactly what it looks like, but you'd scope your current store (the one on RootView) down to a sub-store and a sub-action.

Reply:
And there's one more step: in your `body` just below this you'll need to add a Scope reducer to pass along any actions that this reducer gets to the child reducer.